lib/buffer.js

```diff
@@ -852,6 +852,12 @@ function checkOffset(offset, ext, length) {
     throw new RangeError('Index out of range');
 }
 
+function checkByteLength(byteLength) {
+  if (typeof byteLength !== 'number' || byteLength <= 0)
+    throw new TypeError('"byteLength" must be a positive number ' +
```

Comment:
So should we throw 2 separate errors (`TypeError` when it's not a Number and `RangeError` when `byteLength is <= 0`) or just one `RangeError`?